rpcs3/Emu/Io/Dimensions.cpp

```diff
@@ -384,80 +381,103 @@ u32 dimensions_toypad::load_figure(const std::array<u8, 0x2D * 0x04>& buf, fs::f
 	std::memcpy(figure.data.data(), buf.data(), buf.size());
 	// When a figure is added to the toypad, respond to the game with the pad they were added to, their index,
 	// the direction (0x00 in byte 6 for added) and their UID
-	std::array<u8, 32> figure_change_response = {0x56, 0x0b, figure.pad, 0x00, figure.index, 0x00};
-	std::memcpy(&figure_change_response[6], buf.data(), 7);
+	std::array<u8, 32> figure_change_response = {0x56, 0x0b, figure.pad, 0x00, figure.index, 0x00,
+		buf[0], buf[1], buf[2], buf[4], buf[5], buf[6], buf[7]};
 	figure_change_response[13] = generate_checksum(figure_change_response, 13);
 	m_figure_added_removed_responses.push(figure_change_response);
 
-	if (lock)
-	{
-		m_dimensions_mutex.unlock();
-	}
 	return id;
 }
 
-bool dimensions_toypad::remove_figure(u8 pad, u8 index, bool save, bool lock)
+bool dimensions_toypad::remove_figure(u8 pad, u8 index, bool full_remove)
 {
+	std::lock_guard lock(m_dimensions_mutex);
+
 	dimensions_figure& figure = get_figure_by_index(index);
 	if (figure.index == 255)
 	{
 		return false;
 	}
 
-	if (lock)
```

Comment:
the entire purpose for this was to have a single lock in move_figure, which you now reverted for apparently no reason :P

Reply:
you lock in remove_figure and load_figure, that makes 3 locks in move_figure

Reply:
Yes but the locks for those become out of scope after the method completes it's execution, so they are automatically unlocked. If it's better to only lock and unlock once I can restore that?

Reply:
Not sure if I explained that very well, but previously I was directly calling lock on the mutex in the move method, so in the remove/load methods the flag was needed so that the mutex wasn't locked again.

I have now removed directly locking the mutex in the move method, and then when remove/load are called again the mutex is locked and unlocked there. If it is better to lock in the move method once and keep the flag in the load/remove methods I can restore that change

Reply:
It's not about how many times we lock.
The reason why it should be one lock in move is that we don't want the game or anything else to move/add/remove a figure while we are moving another, which could lead to unexpected behaviour during the move.

Reply:
FYI, I added the lock to move_figure, not you :)

Reply:
Okay that makes sense, I will restore your changes then